File: Data Structures & Algorithms/two-integer-sum/submission-6.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        import numpy as np

        nums = np.array(nums)

        idx_sorted = np.argsort(nums)

        nums = nums[idx_sorted]

        idx_l = 0
        idx_r = len(nums) - 1

        while idx_l < idx_r:
            sum_c = nums[idx_l] + nums[idx_r]
            if sum_c == target:
                a = idx_sorted[idx_l]
                b = idx_sorted[idx_r]
                if a < b:
                    return [a, b]
                else:
                    return [b, a]
            else:
                if sum_c < target:
                    idx_l += 1
                else:
                    idx_r -= 1

File: Data Structures & Algorithms/duplicate-integer/submission-6.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        if (nums.size() <= 1) return false;
        for (auto index_1=0; index_1<=nums.size()-2; index_1++) {
            for (auto index_2=index_1+1; index_2<=nums.size()-1; index_2++) {
                if (nums[index_1] == nums[index_2]) return true;
            }
        }
        return false;
    }
};
